core/lib/main.js

```diff
@@ -136,7 +139,8 @@ async function setup() {
 	app.ws('/start', async (ws, req) => {
 		state.busy();
 
-		const reconnect = parse(req.originalUrl).query === 'reconnect'; // eslint-disable-line
+		const reconnect =
+			new URL(req.originalUrl).search.substring(1) === 'reconnect';
```

Comment:
@roman-mazur I have narrowed down the errors we are getting for Typeerror to this change in my PR. 
Can you recommend what is wrong with it, pinged you on FD